rust/otel-arrow-rust/src/otap/filter/logs.rs

```diff
@@ -0,0 +1,492 @@
+    // against.
+    bodies: Vec<AnyValue>,
+}
+
+/// LogSeverityNumberMatchProperties specifies the requirements needed to match on the log severity field
+#[derive(Debug, Clone, Deserialize)]
+pub struct LogServerityNumberMatchProperties {
+    // Min is the minimum severity needed for the log record to match.
+    // This corresponds to the short names specified here:
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#displaying-severity
+    // this field is case-insensitive ("INFO" == "info")
+    min: i32,
+    // MatchUndefined lets logs records with "unknown" severity match.
+    // If MinSeverity is not set, this field is ignored, as fields are not matched based on severity.
+    match_undefined: bool,
+}
+
+/// LogMatchType describes how we should match the String values provided
+#[derive(Debug, Clone, Deserialize)]
+pub enum LogMatchType {
+    /// match on the string values exactly how they are defined
+    Strict,
+    /// apply string values as a regexp
+    Regexp,
+}
+
+impl LogMatchProperties {
+    /// create a new LogMatchProperties
+    #[must_use]
+    pub fn new(
+        match_type: LogMatchType,
+        resource_attributes: Vec<KeyValue>,
+        record_attributes: Vec<KeyValue>,
+        severity_texts: Vec<String>,
+        severity_number: Option<LogServerityNumberMatchProperties>,
+        bodies: Vec<AnyValue>,
+    ) -> Self {
+        Self {
+            match_type,
+            resource_attributes,
+            record_attributes,
+            severity_texts,
+            severity_number,
+            bodies,
+        }
+    }
+
+    /// create filter takes a logs_payload and returns the filters for each of the record batches, also takes a invert flag to determine if the filters will be inverted
+    pub fn create_filters(
+        &self,
+        logs_payload: &OtapArrowRecords,
+        invert: bool,
+    ) -> Result<(BooleanArray, BooleanArray, BooleanArray)> {
+        let (mut resource_attr_filter, mut log_record_filter, mut log_attr_filter) = (
+            self.get_resource_attr_filter(logs_payload)?,
+            self.get_log_record_filter(logs_payload)?,
+            self.get_log_attr_filter(logs_payload)?,
+        );
+
+        if invert {
+            resource_attr_filter =
+                arrow::compute::not(&resource_attr_filter).expect("not doesn't fail");
+            log_record_filter = arrow::compute::not(&log_record_filter).expect("not doesn't fail");
+            log_attr_filter = arrow::compute::not(&log_attr_filter).expect("not doesn't fail");
+        }
+
+        Ok((resource_attr_filter, log_record_filter, log_attr_filter))
+    }
+
+    fn get_resource_attr_filter(&self, logs_payload: &OtapArrowRecords) -> Result<BooleanArray> {
+        // get resource_attrs record batch
+        let resource_attrs = logs_payload
+            .get(ArrowPayloadType::ResourceAttrs)
+            .context(error::LogRecordNotFoundSnafu)?;
```

Comment:
We discussed on Teams earlier, but just documenting again here so the TODO doesn't get lost. `ResourceAttrs` is an optional `RecordBatch`, so if this record batch doesn't exist, it means that no resources could match the filters. 

I think we could handle this by returning a `BooleanArray` filter mask completely containing false?